parser_library/src/processing/preprocessors/db2_preprocessor.cpp

```diff
@@ -711,12 +716,16 @@ class db2_preprocessor : public preprocessor
 
         generate_replacement(it, end, true);
 
-        return document(std::move(m_result));
+        if (m_source_translated || !m_conditional)
```

Comment:
I guess that when `m_conditional` is `true` we can return right after call to `generate_replacement()`?

Reply:
??? that's what we do here, right?

Reply:
I meant before call to other stuff like `generate_replacement()` and `skip_process()` in order to save some processing time. Or are those functions needed to be called even in this case?

Reply:
Ok, I think I understand now where the confusion is coming from. 

The original comment was related to `generate_replacement(document doc)` and not `generate_replacement(it, end, true);`


In other words - when we enter this function, we can immediately check if `m_conditional` is `true` and return immediately without calling `skip_process(it, end);` and `generate_replacement(it, end, true);`

Reply:
No we can't. We need to go through the document to verify whether there is something to replace.